code/game/objects/items/robot/ai_upgrades.dm

```diff
@@ -18,15 +18,18 @@
 		to_chat(AI, "<span class='userdanger'>[user] has attempted to upgrade you with combat software that you already possess. You gain 50 points to spend on Malfunction Modules instead.</span>")
 	else
 		to_chat(AI, "<span class='userdanger'>[user] has upgraded you with combat software!</span>")
+		to_chat(AI, "<span class='userdanger'>Your current laws and objectives remain unchanged.</span>") //this unlocks malf powers, but does not give the license to plasma flood
 		AI.add_malf_picker()
+		log_game("[key_name(user)] has upgraded [key_name(AI)] with a [src].")
+		message_admins("[ADMIN_LOOKUPFLW(user)] has upgraded [ADMIN_LOOKUPFLW(AI)] with a [src].")
 	to_chat(user, "<span class='notice'>You upgrade [AI]. [src] is consumed in the process.</span>")
 	qdel(src)
 
 
 //Lipreading
 /obj/item/surveillance_upgrade
 	name = "surveillance software upgrade"
-	desc = "A software package that will allow an artificial intelligence to 'hear' from its cameras via lip reading."
+	desc = "An illegal software package that will allow an artificial intelligence to 'hear' from its cameras via lip reading and hidden microphones."
```

Comment:
Spurious change? Anyways how could a software update physically add microphones?

Reply:
The malf upgrade mentions hidden microphones in its desciption, I figured I'd change this one for consistency. 
My best bet is <s>black magic</s> bluespace technology.

https://github.com/tgstation/tgstation/blob/740adc637708f9164217ea4d927004fc4772c473/code/modules/antagonists/traitor/equipment/Malf_Modules.dm#L849